client/app/components/Select.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useMemo } from "react";
+import { maxBy } from "lodash";
+import AntdSelect, { SelectProps } from "antd/lib/select";
+import { calculateTextWidth } from "@/lib/calculateTextWidth";
+
+function SelectWithVirtualScroll({ options, ...props }: SelectProps<any>): JSX.Element {
+  let dropdownMatchSelectWidth: number | boolean = true;
+  useMemo(() => {
+    if (options && options.length > 400) {
+      const largestOpt = maxBy(options, "label.length");
+
+      if (largestOpt) {
+        const offset = 40;
+        const optionText = String(largestOpt.label);
+        const width = calculateTextWidth(optionText);
+        if (width) {
+          dropdownMatchSelectWidth = width + offset; // eslint-disable-line react-hooks/exhaustive-deps
+        }
+      }
+    } else {
+      dropdownMatchSelectWidth = false;
+    }
+  }, [options]);
```

Comment:
```suggestion
  const dropdownMatchSelectWidth = useMemo(() => {
    if (options && options.length > 400) {
      const largestOpt = maxBy(options, "label.length");

      if (largestOpt) {
        const offset = 40;
        const optionText = String(largestOpt.label);
        const width = calculateTextWidth(optionText);
        if (width) {
          return width + offset;
        }
      }
      return false;
  }, [options]);
```

Note: I didn't test this